doc: security: rename trusted_storage (#23888)
nrfconnect/sdk-nrf: doc/nrf/protocols/matter/end_product/security.rst

By default, the DAC private key is stored in the KMU storage while using TF-M.
See the :ref:`matter_platforms_security_dac_priv_key_kmu` section for more information.

Secure storage
**************

> The :ref:`secure_storage_in_ncs` is a security mechanism designed to securely store and manage sensitive data.

Reviewer: ```suggestion
The :ref:`secure_storage_in_ncs` is a security mechanism for securely storing and managing sensitive data.
```
